core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestUtils.java

```diff
@@ -730,6 +737,71 @@ public static Optional<RateLimiter> createRateLimiter(long rate) {
     return Optional.of(RateLimiter.create(rate));
   }
 
+  /**
+   * Initiate the S3 API metadata directories.
+   * @param fileSystem
+   * @return true if directory is created or already exists
+   */
+  private static boolean initMultipartUploadsMetadataDir(FileSystem fileSystem) {
+    try {
+      AlluxioURI path = new AlluxioURI(MULTIPART_UPLOADS_METADATA_DIR);
+      if (!fileSystem.exists(path)) {
+        LOG.info("Create multipart uploads metadata dir: {}", MULTIPART_UPLOADS_METADATA_DIR);
+        fileSystem.createDirectory(
+            path,
+            CreateDirectoryPOptions.newBuilder()
+                .setRecursive(true)
+                .setMode(PMode.newBuilder()
+                    .setOwnerBits(Bits.ALL)
+                    .setGroupBits(Bits.ALL)
+                    .setOtherBits(Bits.NONE).build())
+                .setWriteType(getS3WriteType())
+                .setXattrPropStrat(XAttrPropagationStrategy.LEAF_NODE)
+                .build()
+        );
+      }
+      return true;
+    } catch (Exception e) {
+      // If we only use read API or mount root as read-only, this directory is not necessary,
+      // so we just print the error log instead of throwing exception here.
+      LOG.error("Can not init multipart uploads metadata dir: {}", MULTIPART_UPLOADS_METADATA_DIR,
+          e);
+      return false;
+    }
+  }
+
+  /**
+   * Try to create S3 API metadata directories.
+   * @param context
+   * @param fileSystem
+   */
+  @SuppressWarnings("unchecked")
+  public static void tryInitMultipartUploadsMetadataDir(ServletContext context,
+      FileSystem fileSystem) {
+    AtomicReference<Boolean> flag = (AtomicReference<Boolean>) context.getAttribute(
+        ProxyWebServer.MULTIPART_UPLOADS_METADATA_DIR_CREATE_FLAG);
+    if (flag.get() == null) {
+      // If flag is null, it means the directory has never been created.
+      // For the first time, we should lock all requests to ensure that the directory is created.
+      synchronized (S3RestUtils.class) {
+        // Double check lock.
+        if (flag.get() == null) {
+          flag.set(S3RestUtils.initMultipartUploadsMetadataDir(fileSystem));
+        }
+      }
+    } else {
+      // If flag is ture, it means that the directory has been created, we should do nothing.
+      // If flag is false, it means that the directory creation failed and need to retry.
+      // Make sure that only one request attempts to create the directory at the same moment,
+      // and other requests quickly pass through.
+      if (flag.compareAndSet(false, true)) {
```

Comment:
And even if the creation fails here, it does not seem to affect the execution of the requests. I'm not sure if it's good or bad. What do you think of this problem? @lucyge2022

Reply:
> Because sometimes creating a directory will never succeed. like https://github.com/Alluxio/alluxio/issues/17160. 

agreed, as current PR is only to get MultipartUpload out of the way and make it a user responsibility. This folder is a merely a replacement of master responsibility to record a state in a db. Even if its used in this way, it should be treated as a system-user accessed only where the metaFs should have full access to it. Would very much love to keep this PR suspended for now, and get these problem solved in the right way.